### PR TITLE
Fix/3809 No reminder to share a positive test result if user first register QR code before test result positive

### PR DESCRIPTION
**Pull request by SwissMountainFrog**
This PR fixes an issue if the test result changes while the user has the app in the background (so the test result was `pending` and changed to `positive` on the server.

Also, this PR fixes an issue with "updating" an existing `pending` result and the updated state is `positive`.

Now in both situations, the two "warn others reminder" get scheduled as expected.

The PR is [based on this JIRA Ticket](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-3809)

### CODE REVIEW

**Review by nickguendling — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultViewModel.swift [hunk 1]
```diff
@@ -125,7 +125,7 @@ class ExposureSubmissionTestResultViewModel {
 	}
 	
 	func updateWarnOthers() {
-		self.warnOthersReminder.evaluateNotificationState(testResult: self.testResult)
+		warnOthersReminder.evaluateNotificationState(testResult: testResult)
 	}
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultViewModel.swift:129]: inside the prive section - make it private too or move?

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultViewModel.swift [hunk 2]
```diff
@@ -163,6 +163,7 @@ class ExposureSubmissionTestResultViewModel {
 				self?.error = error
 			case let .success(testResult):
 				self?.testResult = testResult
+				self?.updateWarnOthers()
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultViewModel.swift:166]: if this is the only caller why use a separte func?

**Review by KaiTeuber — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultViewModel.swift (hunk 1 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultViewModel.swift:129]: Moved. Thanks.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultViewModel.swift (hunk 2 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultViewModel.swift:166]: The reason is to keep it logically separated

**Review by KaiTeuber — APPROVED** [review]

**Pull request merged**